internal/addon/authentication/provider.go

```diff
@@ -17,50 +17,43 @@ import (
 // AuthenticationType defines the type of authentication that will be used for a target.
 type AuthenticationType string
 
-type ProviderConfig struct {
+type Config struct {
 	StaticAuthConfig manifests.StaticAuthenticationConfig
 	MTLSConfig       manifests.MTLSConfig
 }
 
 // secretsProvider is a struct that holds the Kubernetes client and configuration.
 type secretsProvider struct {
+	ctx         context.Context
```

Comment:
Why is this context needed here? Often contexts are passed around functions (we could add it in `GenerateSecrets` and `FetchSecrets`) and not stored in structs (see https://go.dev/blog/context-and-structs).